app/services/publish_draft_edition_service.rb

```diff
@@ -73,9 +73,12 @@ def set_new_live_edition
     document.reload_live_edition
   end
 
-  def set_first_published_at
+  def set_published_at
+    current_time = Time.current
+    edition.update!(published_at: current_time)
```

Comment:
Rather than running an `update!` that calls an instant query we can just assign the attribute with `edition.published_at = current_time` and save an unnecessary query (we're going to run a save later.

Reply:
Good point, done!